src/execution/mod.rs

```diff
@@ -328,6 +333,191 @@ where
+                    let v1: i64 = stack.pop_value(ValType::NumType(NumType::I64)).into();
+                    let res = v1.wrapping_sub(v2);
+
+                    trace!("Instruction: i64.sub [{v1} {v2}] -> [{res}]");
+                    stack.push_value(res.into());
+                }
+                I64_MUL => {
+                    let v1: i64 = stack.pop_value(ValType::NumType(NumType::I64)).into();
+                    let v2: i64 = stack.pop_value(ValType::NumType(NumType::I64)).into();
+                    let res = v1.wrapping_mul(v2);
+
+                    trace!("Instruction: i64.mul [{v1} {v2}] -> [{res}]");
+                    stack.push_value(res.into());
+                }
+                I64_DIV_S => {
+                    let dividend: i64 = stack.pop_value(ValType::NumType(NumType::I64)).into();
+                    let divisor: i64 = stack.pop_value(ValType::NumType(NumType::I64)).into();
+
+                    if dividend == 0 {
+                        return Err(RuntimeError(DivideBy0));
+                    }
+                    if divisor == i64::MIN && dividend == -1 {
+                        return Err(RuntimeError(UnrepresentableResult));
+                    }
+
+                    let res = divisor / dividend;
+
+                    trace!("Instruction: i64.div_s [{divisor} {dividend}] -> [{res}]");
+                    stack.push_value(res.into());
+                }
+                I64_DIV_U => {
+                    let dividend: i64 = stack.pop_value(ValType::NumType(NumType::I64)).into();
+                    let divisor: i64 = stack.pop_value(ValType::NumType(NumType::I64)).into();
+
+                    let dividend = dividend as u64;
+                    let divisor = divisor as u64;
+
+                    if dividend == 0 {
+                        return Err(RuntimeError(DivideBy0));
+                    }
+
+                    let res = (divisor / dividend) as i64;
+
+                    trace!("Instruction: i64.div_u [{divisor} {dividend}] -> [{res}]");
+                    stack.push_value(res.into());
+                }
+                I64_REM_S => {
+                    let dividend: i64 = stack.pop_value(ValType::NumType(NumType::I64)).into();
+                    let divisor: i64 = stack.pop_value(ValType::NumType(NumType::I64)).into();
+
+                    if dividend == 0 {
+                        return Err(crate::Error::RuntimeError(
+                            crate::core::error::RuntimeError::DivideBy0,
+                        ));
+                    }
+
+                    let res = divisor.checked_rem(dividend);
+                    let res = res.unwrap_or_default();
+
+                    trace!("Instruction: i64.rem_s [{divisor} {dividend}] -> [{res}]");
+                    stack.push_value(res.into());
+                }
+                I64_REM_U => {
+                    // this is a correct implementation
```

Comment:
I would rather have the reasoning in the comment than an implementation link that does the same

Reply:
I removed it altogether, I assumed it was too trivial to have it.